data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v3/validator/GetNewBlockV3.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.ValidatorDataProvider;
+import tech.pegasys.teku.api.schema.Version;
+import tech.pegasys.teku.bls.BLSSignature;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.datastructures.blocks.BlockContainer;
+import tech.pegasys.teku.spec.datastructures.metadata.BlockContainerAndMetaData;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionCache;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitions;
+import tech.pegasys.teku.storage.client.ChainDataUnavailableException;
+
+public class GetNewBlockV3 extends RestApiEndpoint {
+
+  public static final String ROUTE = "/eth/v3/validator/blocks/{slot}";
+
+  protected final ValidatorDataProvider provider;
+
+  public GetNewBlockV3(
+      final DataProvider dataProvider, final SchemaDefinitionCache schemaDefinitionCache) {
+    this(dataProvider.getValidatorDataProvider(), schemaDefinitionCache);
+  }
+
+  public GetNewBlockV3(
+      final ValidatorDataProvider provider, final SchemaDefinitionCache schemaDefinitionCache) {
+    super(getEndpointMetaData(schemaDefinitionCache));
+    this.provider = provider;
+  }
+
+  private static EndpointMetadata getEndpointMetaData(
+      final SchemaDefinitionCache schemaDefinitionCache) {
+    return EndpointMetadata.get(ROUTE)
+        .operationId("produceBlockV3")
+        .summary("Produce a new block, without signature")
+        .description(
+            "Requests a beacon node to produce a valid block, which can then be signed by a validator. The\n"
+                + "returned block may be blinded or unblinded, depending on the current state of the network as\n"
+                + "decided by the execution and beacon nodes.\n"
+                + "The beacon node must return an unblinded block if it obtains the execution payload from its\n"
+                + "paired execution node. It must only return a blinded block if it obtains the execution payload\n"
+                + "header from an MEV relay.\n"
+                + "Metadata in the response indicates the type of block produced, and the supported types of block\n"
+                + "will be added to as forks progress.")
+        .tags(TAG_VALIDATOR, TAG_VALIDATOR_REQUIRED)
+        .pathParam(SLOT_PARAMETER.withDescription(SLOT_PATH_DESCRIPTION))
+        .queryParamRequired(RANDAO_PARAMETER)
+        .queryParam(GRAFFITI_PARAMETER)
+        .queryParam(SKIP_RANDAO_VERIFICATION_PARAMETER)
```

Comment:
`SKIP_RANDAO_VERIFICATION_PARAMETER` is also present in V2 but apparently we never supported it.
Indeed we haven't implemented the logic, and in fact you don't read the param in `handleRequest`.
But we could leave it as a separate PR.

Reply:
Yep, I'll do that in a separate PR